include/z64.h

```diff
@@ -969,6 +969,80 @@ typedef struct {
     /* 0x4C */ u32 unk_4C;
 } PreRender; // size = 0x50
 
+#define TRANS_TRIGGER_OUT -20 // transition is going out
+#define TRANS_TRIGGER_OFF 0 // transition is not active
+#define TRANS_TRIGGER_IN 20 // activate transition to come in
+
+typedef enum {
+    /* 00 */ TRANS_MODE_OFF,
+    /* 01 */ TRANS_MODE_SETUP,
+    /* 02 */ TRANS_MODE_INSTANCE_INIT,
+    /* 03 */ TRANS_MODE_INSTANCE_RUNNING,
```

Comment:
What's the motivation for putting these transition macros & enums in `z64.h`? Could they go in `z64transition.h` instead?

Reply:
transition.h is specifically for the "instance" type transitions. it didnt really feel right to, these are more about the global context facing variables

Reply:
imo z64transition will get a different name at some point to be more specific about those kinds of transitions, and the TransitionContext + these enums will go in the actual transition.h

Reply:
well actually no, cause the transition context is unique to the instance types... idk its complicated lol. but yeah we need to put more thought into that kinda thing in general. so z64.h is fine for now